move-vm-support/src/base58_address.rs

```diff
@@ -0,0 +1,183 @@
+    if decoded_base58.len() != BASE58_LEN {
+        bail!(
+            "unsupported base58 address length (base58): {}",
+            decoded_base58.len()
+        );
+    }
+
+    AccountAddress::from_bytes(decoded_base58).map_err(anyhow::Error::msg)
+}
+
+/// Convert Base58 address to Move address string.
+pub fn base58_to_move_address_string(base58: &str) -> Result<String> {
+    Ok(format!("{:#X}", base58_to_move_address(base58)?))
+}
+
+/// Convert Move address to base58 address string
+/// Read more about the address structure: https://docs.substrate.io/reference/address-formats/
+///
+/// For now this function is supporting only one-byte address types, with one universal address type
+/// for all Move accounts and it's 42. We can't do otherwise as we've no information about the
+/// network we're currently compiling for.
+///
+/// Move developers should be aware, that information about any fails during the compilation of
+/// Move code will point to the addresses converted to the 42 address type.
+/// ```
+/// use move_core_types::account_address::AccountAddress;
+/// use move_vm_support::base58_address::move_address_to_base58_string;
+/// let move_address = "0x02A212DE6A9DFA3A69E22387ACFBAFBB1A9E591BD9D636E7895DCFC8DE05F331";
+/// let substrate_address = move_address_to_base58_string(&AccountAddress::from_hex_literal(move_address).unwrap());
+/// assert_eq!(
+///   "BH9wjrYwKi76DbU6LC1X4VEeevUTkaKgxGnaFNFaqZA",
+///  substrate_address
+/// );
+/// ```
+pub fn move_address_to_base58_string(addr: &AccountAddress) -> String {
+    let mut base58_address = [0; BASE58_LEN];
+    base58_address[0..PUB_KEY_LEN].copy_from_slice(&addr.to_vec());
+    bs58::encode(base58_address).into_string()
+}
+
+/// Converts a base58 address string to a Substrate SS58 address string.
+pub fn base58_string_to_ss588_string(addr: &str) -> Result<String> {
+    let move_address = base58_to_move_address(addr)?;
+    Ok(crate::ss58_address::move_address_to_ss58_string(
+        &move_address,
+    ))
+}
+
+/// Converts a Substrate SS58 address string into a base58 string.
+pub fn ss58_string_to_base58_string(addr: &str) -> Result<String> {
+    let move_address = crate::ss58_address::ss58_to_move_address(addr)?;
+    Ok(move_address_to_base58_string(&move_address))
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_base58_to_move_correct() {
+        let substrate_address_base58 = "BH9wjrYwKi76DbU6LC1X4VEeevUTkaKgxGnaFNFaqZA";
+        let move_address = base58_to_move_address_string(substrate_address_base58).unwrap();
+
+        assert_eq!(
+            (move_address.len() - 2) / 2, // 2 hex chars per byte
+            PUB_KEY_LEN
+        );
+
+        assert_eq!(
+            "0x02A212DE6A9DFA3A69E22387ACFBAFBB1A9E591BD9D636E7895DCFC8DE05F331",
+            move_address
+        );
+
+        let substrate_address = "7s3xJXvnK3hZCE1ZvuHyhUxifWbQdiLa3QffBccBaQy";
```

Comment:
Is this one also `substrate_address_base58`? Maybe I would just call these `base58_address` and that's it
```suggestion
        let substrate_address_base58 = "7s3xJXvnK3hZCE1ZvuHyhUxifWbQdiLa3QffBccBaQy";
```

Reply:
Done